node/pkg/watchers/sui/watcher.go

```diff
@@ -239,184 +250,222 @@ func (e *Watcher) Run(ctx context.Context) error {
-				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-				continue
-			}
+				var res SuiCheckpointSN
+				err = json.Unmarshal(body, &res)
+				if err != nil {
+					logger.Error("unmarshal failed into uint64", zap.String("body", string(body)), zap.Error(err))
+					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+					return fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to unmarshal body: %s, error: %w", string(body), err)
+				}
 
-			body, err := io.ReadAll(resp.Body)
-			if err != nil {
-				logger.Error("unexpected truncated body when calling getEvents", zap.Error(err))
-				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-				continue
+				height, pErr := strconv.ParseInt(res.Result, 0, 64)
+				if pErr != nil {
+					logger.Error("Failed to ParseInt")
+				} else {
+					currentSuiHeight.Set(float64(height))
+					logger.Debug("sui_getLatestCheckpointSequenceNumber", zap.String("result", res.Result))
+
+					p2p.DefaultRegistry.SetNetworkStats(vaa.ChainIDSui, &gossipv1.Heartbeat_Network{
+						Height:          int64(height),
+						ContractAddress: e.suiAccount,
+					})
+				}
 
+				readiness.SetReady(e.readinessSync)
 			}
-			resp.Body.Close()
+		}
+	})
 
-			logger.Info("receive", zap.String("body", string(body)))
+	common.RunWithScissors(ctx, errC, "sui_fetch_obvs_req", func(ctx context.Context) error {
+		for {
+			select {
+			case <-ctx.Done():
+				logger.Error("sui_fetch_obvs_req context done")
+				return ctx.Err()
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDSui {
+					panic("invalid chain ID")
+				}
 
-			var res SuiTxnQuery
-			err = json.Unmarshal(body, &res)
-			if err != nil {
-				logger.Error("failed to unmarshal event message", zap.Error(err))
-				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-				continue
+				id := base64.StdEncoding.EncodeToString(r.TxHash)
 
-			}
+				logger.Info("obsv request", zap.String("TxHash", string(id)))
 
-			for _, chunk := range res.Result.Data {
-				err := e.inspectBody(logger, chunk)
-				if err != nil {
-					logger.Error("unspected error while parsing chunk data in event", zap.Error(err))
-				}
-			}
-		case msg := <-pumpData:
-			logger.Info("receive", zap.String("body", string(msg)))
-
-			var res SuiEventMsg
-			err = json.Unmarshal(msg, &res)
-			if err != nil {
-				logger.Error("Failed to unmarshal SuiEventMsg", zap.String("body", string(msg)), zap.Error(err))
-				continue
-			}
-			if res.Error != nil {
-				_ = ws.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
-				return errors.New((*res.Error).Message)
-			}
-			if res.ID != nil {
-				if *res.ID == e.subId {
-					logger.Debug("Subscribed set to true")
-					e.subscribed = true
-				}
-				continue
-			}
+				buf := fmt.Sprintf(`{"jsonrpc":"2.0", "id": 1, "method": "sui_getEvents", "params": [{"Transaction": "%s"}, null, 10, true]}`, id)
```

Comment:
What are the `params` here? I tried looking at https://docs.sui.io/sui-jsonrpc#sui_getEvents but it doesn't quite line up.